community/samples/serving/helloworld-java-quarkus/README.md

```diff
@@ -0,0 +1,452 @@
+                 <goal>add-resource</goal>
+               </goals>
+               <configuration>
+                 <resources>
+                   <resource>
+                     <directory>${project.basedir}/src/main/knative</directory>
+                     <targetPath>${project.build.directory}/knative</targetPath>
+                     <filtering>true</filtering>
+                   </resource>
+                 </resources>
+               </configuration>
+             </execution>
+           </executions>
+         </plugin>
+       </plugins>
+     </build>
+     <dependencies>
+       <dependency>
+         <groupId>io.quarkus</groupId>
+         <artifactId>quarkus-resteasy</artifactId>
+       </dependency>
+       <dependency>
+         <groupId>io.quarkus</groupId>
+         <artifactId>quarkus-junit5</artifactId>
+         <scope>test</scope>
+       </dependency>
+       <dependency>
+         <groupId>io.rest-assured</groupId>
+         <artifactId>rest-assured</artifactId>
+         <scope>test</scope>
+       </dependency>
+     </dependencies>
+     <profiles>
+       <profile>
+         <id>native</id>
+         <build>
+           <plugins>
+             <plugin>
+               <groupId>io.quarkus</groupId>
+               <artifactId>quarkus-maven-plugin</artifactId>
+               <version>${quarkus.version}</version>
+               <executions>
+                 <execution>
+                   <goals>
+                     <goal>native-image</goal>
+                   </goals>
+                   <configuration>
+                     <enableHttpUrlHandler>true</enableHttpUrlHandler>
+                   </configuration>
+                 </execution>
+               </executions>
+             </plugin>
+           </plugins>
+         </build>
+       </profile>
+     </profiles>
+   </project>
+   ```
+
+1. Run the application locally:
+
+   ```shell
+   ./mvnw compile quarkus:dev
+   ```
+
+   Go to `http://localhost:8080/` to see your `Hello World!` message.
+
+1. In your project directory, create a file named `Dockerfile` and copy the code
+   block below into it.
+
+   ```docker
+   FROM quay.io/rhdevelopers/quarkus-java-builder:graal-1.0.0-rc15 as builder
+   COPY . /project
+   WORKDIR /project
+   # uncomment this to set the MAVEN_MIRROR_URL of your choice, to make faster builds
+   # ARG MAVEN_MIRROR_URL=<your-maven-mirror-url>
+   # e.g.
+   #ARG MAVEN_MIRROR_URL=http://192.168.64.1:8081/nexus/content/groups/public
+
+   RUN /usr/local/bin/entrypoint-run.sh mvn -DskipTests clean package
+
+   FROM fabric8/java-jboss-openjdk8-jdk:1.5.4
+   USER jboss
+   ENV JAVA_APP_DIR=/deployments
+
+   COPY --from=builder /project/target/lib/* /deployments/lib/
+   COPY --from=builder /project/target/*-runner.jar /deployments/app.jar
+
+   ENTRYPOINT [ "/deployments/run-java.sh" ]
+   ```
+
+   If you want to build Quarkus native image, then copy the following code block in to file called `Dockerfile.native`
```

Comment:
This is not a standalone element in the list and starts with "If", does that mean its optional?